src/utils/smallbuffer.h

```diff
@@ -0,0 +1,68 @@
+/*! \file
+ * \brief A container for arbitrary bytes.
+ * \details Tracks capacity, supports small buffer optimization.
+ * \copyright MIT license.
+ * \date 2025
+ * \author Magnus Larsen
+ */
+
+#ifndef SMALLBUFFER_H
+#define SMALLBUFFER_H
+
+#include <stddef.h>
+#include <string.h>
+
+typedef struct smallbuffer {
+    // private fields, do not access.
+    ptrdiff_t priv_cap;
+    union {
+        char *priv_alloc;
+        char priv_inline[sizeof(char *)];
```

Comment:
Why is this `sizeof(char *)`, basically the size of a pointer, and not something like `8`?